java/src/main/java/com/logicalclocks/base/engine/FeatureViewEngineBase.java

```diff
@@ -0,0 +1,279 @@
+/*
+ *  Copyright (c) 2022. Hopsworks AB
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ *  See the License for the specific language governing permissions and limitations under the License.
+ *
+ */
+
+package com.logicalclocks.base.engine;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.logicalclocks.base.Feature;
+import com.logicalclocks.base.constructor.Join;
+import com.logicalclocks.base.constructor.QueryBase;
+import com.logicalclocks.base.metadata.FeatureGroupBase;
+import com.logicalclocks.base.metadata.FeatureViewApi;
+import com.logicalclocks.base.metadata.TagsApi;
+import com.logicalclocks.base.EntityEndpointType;
+import com.logicalclocks.base.FeatureStoreBase;
+import com.logicalclocks.base.FeatureStoreException;
+import com.logicalclocks.base.FeatureViewBase;
+import com.logicalclocks.base.TrainingDatasetFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FeatureViewEngineBase {
```

Comment:
this class should probably be made into an abstract class. Furthermore, even though `FeatureViewEngine` extends `FeatureViewEngineBase` there is a huge amount of duplication between the two.

Reply:
There is still a lot of duplication between the two (for ex. `get` methods). Furthermore, I think we should define all public methods as abstract also in the base class, just so we don't forget to add it later on (ex. `getBatchData`).